app/models/enterprise_group.rb

```diff
@@ -28,4 +39,50 @@ class EnterpriseGroup < ActiveRecord::Base
 
   scope :by_position, order('position ASC')
   scope :on_front_page, where(on_front_page: true)
+  scope :managed_by, lambda { |user|
+    if user.has_spree_role?('admin')
+      scoped
+    else
+      where('owner_id = ?', user.id);
+    end
+  }
+
+  def set_unused_address_fields
+    address.firstname = address.lastname = 'unused' if address.present?
+  end
+
+  def set_undefined_address_fields
+    if !address.present?
+      return
+    end
+    address.phone.present? || address.phone = 'undefined'
+    address.address1.present? || address.address1 = 'undefined'
+    address.city.present? || address.city = 'undefined'
+    address.zipcode.present? || address.zipcode = 'undefined'
+  end
+
+  def phone
```

Comment:
Want to play with some metaprogramming? We can express all of these methods in one go using `define_method`. It's a little more magical but tidies this up a bit.

But then there's the question of whether we should be using this pattern in the first place... for discussion.